### PR TITLE
4972 - Implement the ids-rating web comoponent

### PR DESCRIPTION
**Pull request by danielortiz1982**
**Explain the details for making this change. What existing problem does the pull request solve?**

**Related github/jira issue (required)**:
Ids Rating web component 5x. A user can select a rating by clicking the star with the position that corresponds to the users rating out of the total. The selected stars state will change in color and the value attribute will update

**Steps necessary to review your pull request (required)**:
Include:
- git checkout 4972-ids-rating-wc
- Pull, Build, Run!
- Test coverage `npm run test test/ids-rating/ids-rating-func-test.js`
- Click on any given star for the state change and updated attribute values.

**Full star whole values** 
image
image

**Half star decimal .5 values**
image

- Axe Dev Tools
Screen Shot 2021-05-21 at 7 26 50 AM

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Comment by tmcconechy** [comment]
@danielortiz1982 - just remove the WIP and add a PR description (see some others for reference) at that point I / we will review the PR

File: src/ids-rating/README.MD [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+## Variants
+The rating component can be used as an interactive or read only element. Users may select any whole value out of the total as a rating. When functioning as a read-only element, ratings may use .5 decimals to more accurately display the average rating of an object.
+
+```html
+<ids-rating stars="5" readonly="true" value="4.5" size="large"></ids-rating>
+```
+
+![Variants Image](https://ids-com.s3.amazonaws.com/documents/rating_variant.svg)
+
+## States
```
Comment (tmcconechy) [inline src/ids-rating/README.MD:30]: I dont think we need to include these static images as they will be out of date. So i would suggest removing the variants section and just mention the "decimals" version in the states or show a code example for it in the above section.

File: src/ids-rating/README.MD [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+## Settings and Attributes
+- `stars` {number} Set the stars attribute to displayed the amount of items`stars="5"`
+- `value` {number} Set the rating value attribute. `value="0"`
+- `readyonly` {boolean} Set the readonly attribute. Valid values are 'true' | 'false'
+- `size` {number} Set the rating size attribute. `size="large"` Valid values are 'small' | 'medium' | 'large'
+- `mode` {string} Set the theme mode
+- `version` {string} Set the theme version
+
+## Designs
```
Comment (tmcconechy) [inline src/ids-rating/README.MD:49]: Can remove this section its already out of date

File: src/ids-rating/TODO.MD [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# TODO on IDS Rating
+
+## Features
+
+- [] Stars attribute render stars on setter and getter
```
Comment (tmcconechy) [inline src/ids-rating/TODO.MD:5]: Can we merge the CHECKLIST.md with the TODO.md into one (just TODO.md) as they seem redundant.

File: src/ids-rating/ids-rating.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+  mix,
+  stringUtils,
+  props
+} from '../ids-base/ids-element';
+
+import { IdsEventsMixin } from '../ids-base/ids-events-mixin';
+import { IdsKeyboardMixin } from '../ids-base/ids-keyboard-mixin';
+import { IdsThemeMixin } from '../ids-base/ids-theme-mixin';
+
+// @ts-ignore
```
Comment (tmcconechy) [inline src/ids-rating/ids-rating.js:14]: I think all ts-ignores can now safely be removed

File: src/ids-rating/ids-rating.js [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+      }
+      html += '</div>';
+      return html;
+    }
+
+    /**
+     * @returns {Array<string>} this component's observable properties
+     */
+    static get properties() {
+      return [...props.MODE, props.VERSION, props.VALUE, 'stars', props.READONLY, props.CLICKABLE, props.COMPACT, props.SIZE];
```
Comment (tmcconechy) [inline src/ids-rating/ids-rating.js:60]: Minor but i started to alphabetize these

File: src/ids-rating/ids-rating.js [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+      let html = '<div id="rating">';
+      for (let i = 0; i < this.stars; i++) {
+        html += `<ids-icon class="star star-${i}" role-"button" icon="star-outlined" tabindex="0" size="${this.size}"></ids-icon><span class="audible">${i + 1} out of 5 Stars</span>`;
+      }
+      html += '</div>';
+      return html;
+    }
+
+    /**
+     * @returns {Array<string>} this component's observable properties
```
Comment (tmcconechy) [inline src/ids-rating/ids-rating.js:57]: Try and add brief comment to all methods

File: src/ids-rating/ids-rating.scss [hunk 7]
```diff
@@ -0,0 +1,55 @@
+@import '../ids-base/ids-base';
+
+#rating {
```
Comment (tmcconechy) [inline src/ids-rating/ids-rating.scss:3]: Is it going to be a problem that the ID is hard coded? I think i would make it class="rating" instead in the template

**Review by tmcconechy — CHANGES_REQUESTED** [review]
See small comments this is looking very good now....but needs some final touch ups

1. Remove draft status?
1. Edit the PR and fill in the template 
1. Either fix the coverage or add /* istanbul ignore next */ to the sections that dropped the coverage if you cant figure it out?
1. add the e2e and percy tests like the others (should take only a few moments)

**Review by tmcconechy — APPROVED** [review]

**Review by EdwardCoyle — APPROVED** [review]
Looks great, well done!

**Comment by tmcconechy** [comment]
Great! Will merge so i can fix my other PR with these changes

**Pull request merged**